assets/js/blocks/reviews/reviews-by-product/no-reviews-placeholder.tsx

```diff
@@ -3,18 +3,28 @@
  */
 import { __, sprintf } from '@wordpress/i18n';
 import { Placeholder, Spinner } from '@wordpress/components';
-import PropTypes from 'prop-types';
-import ErrorPlaceholder from '@woocommerce/editor-components/error-placeholder';
+import ErrorPlaceholder, {
+	ErrorObject,
+} from '@woocommerce/editor-components/error-placeholder';
 import { Icon, commentContent } from '@wordpress/icons';
 import { withProduct } from '@woocommerce/block-hocs';
 import { decodeEntities } from '@wordpress/html-entities';
+/**
+ * Internal dependencies
+ */
+import type { NoReviewsPlaceholderProps } from './types';
 
-const NoReviewsPlaceholder = ( { error, getProduct, isLoading, product } ) => {
+const NoReviewsPlaceholder = ( {
+	error,
+	getProduct,
+	isLoading,
+	product,
+}: NoReviewsPlaceholderProps ) => {
 	const renderApiError = () => (
 		<ErrorPlaceholder
 			className="wc-block-featured-product-error"
-			error={ error }
-			isLoading={ isLoading }
+			error={ error as ErrorObject }
+			isLoading={ isLoading as boolean }
 			onRetry={ getProduct }
 		/>
```

Comment:
What are your thoughts on moving the TS definitions of `error` and `isLoading` into the `ErrorPlaceholder` component? That way, the TS definition lives in a central place and each section that consumes that component is typed by default. Otherwise, we might have to use both `error={ error as ErrorObject }` and `isLoading={ isLoading as boolean }` at various places.

Reply:
@nielslange 
I added this because in [types.ts](https://github.com/woocommerce/woocommerce-blocks/pull/9692/files#diff-124611adde9284bc10a596ab932e40039d1f8aa5fa30054cd71bda768651c803R28)'s NoReviewsPlaceholderProps I made these properties optional so not it expects `undefined` or ErrorObject in case of error.
and I made them optional is while resolving noReviewsPlaceholder ... but it seems I forgot to revert this, as noReviewsPlaceholder is flagged as ts error anyway rn, we can revert this.

Reply:
Pushed changes